app/src/main/java/com/example/mytestproject/util/CityFilter.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.mytestproject.util
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Log
+import com.example.data.utils.getCityModelsListFromJson
+import com.example.domain.models.CityModel
+import com.google.gson.Gson
+import javax.inject.Inject
+
+class CityFilter @Inject constructor(private val context: Context) {
+    private val cityList = mutableListOf<CityModel>()
+
+    init {
+        cityList.addAll(getCityModelsListFromJson(context))
+    }
+
+    fun filterCityList(inputText: String): List<CityModel> {//  method filters the list of all cities by user-entered characters
+        val list = mutableListOf<CityModel>()
+        list.addAll( cityList.filter { it.city_name.contains(inputText, true) })
+        saveFilteredCityList(list)
+        return list
+    }
+
+    private fun saveFilteredCityList(filteredCityList: List<CityModel>) { // method saves filtered list of cities to SharedPreferences
+        val jsonCityList = Gson().toJson(filteredCityList)
```

Comment:
Gson should be injected via class constructor.

Reply:
Fixed